fe/lib/components/LocationSuggest/LocationSuggest.tsx

```diff
@@ -139,6 +145,36 @@ export const LocationSuggest = forwardRef<HTMLInputElement, Props>(
       getLocationSuggest,
     ]);
 
+    const loadInitialLocation = useCallback(async () => {
+      if (!initialValue || initialLocation?.id.value === initialValue) {
+        return;
+      }
+
+      if (!client) {
+        return;
+      }
+      const { data } = await client.query({
+        ...(client && { client }),
+        fetchPolicy: 'no-cache',
+        query: LOCATION,
+        variables: { id: initialValue },
+      });
+
+      if (!data.location) {
+        return;
+      }
+
+      setInitialLocation(data.location);
```

Comment:
Hmm, I'm now thinking my suggestion may have been misguided. What do you think about this:

1. Scrap `initialLocation` and `setInitialLocation` altogether. I think we only needed that state to plumb between the two effects previously.
2. Change earlier guard clause from `initialLocation?.id.value === initialValue` to `selectedLocationId`. If the user has already made a selection I think the initial location no longer matters and we shouldn't need to run the query at all.

The function as-is is going to rerun the query on every `selectedLocationId` change, which is not ideal.

Reply:
HI Ryan,

I am passing `initialLocation` to the component. Are you saying don't pass the whole location object and just pass the id?

```
<LocationSuggestInput
            {...restProps}
            isLoading={nearestLocationsLoading}
            locationSuggestions={locationSuggestResults}
            onChange={setLocationSuggestInput}
            onClear={() => {
              setSelectedLocationId('');
              setPlaceholder('');
            }}
            onDetectLocation={handleDetectLocationClicked}
            onSelect={handleLocationSelect}
            placeholder={placeholder}
            setDetectLocationError={setDetectLocationError}
            tone={tone}
            initialLocation={initialLocation}
          />
          ```

Reply:
Ah my bad, I see you're propagating the contextual name too. I think we just need to extend the initial guard clause then:

```diff
- if (!initialValue || initialLocation?.id.value === initialValue) {
+ if (!initialValue || initialLocation?.id.value === initialValue || selectedLocationId) {
  return;
}